thunder/executors/torch_autograd.py

```diff
@@ -189,13 +191,14 @@ def connect_to_autograd(
     saved_tensors,
     saved_other,
     return_none_instead_of_grads,
+    disable_split_autograd,
 ):
     # PyTorch seems to not like our side channel trick when capturing graphs
     # through dynamo and using cuda graphs.
     # Of course, the real trick is to use the CUDAGraphTransform instead
     # of having something else apply it while introducing funny additional
     # conditions for success.
-    if not is_cudagraph_capturing():
+    if not is_cudagraph_capturing() and not disable_split_autograd:
```

Comment:
Can't use `get_compile_option` here as this part is called where compile_data `cd` is not set.